apps/backend/src/redis.ts

```diff
@@ -0,0 +1,146 @@
+import { Member, Room, Status, WebSocketResponse } from '@dundring/types';
+import { error, hours, isSuccess, success } from '@dundring/utils';
+import { createClient } from 'redis';
+import * as websocket from './websocket';
+
+require('dotenv').config();
+
+const roomChannelPrefix = 'room';
+
+const redisUrl = process.env.REDIS_URL || 'redis://127.0.0.1:6379';
+const redisClient = createClient({
+  url: redisUrl,
+});
+const redisPublisher = redisClient.duplicate();
+const redisSubscriber = redisClient.duplicate();
+redisClient.on('error', (err) => console.log('Redis Client Error', err));
+
+export const initRedis = async () => {
+  console.log('Connect redis client');
+  await Promise.all([
+    redisClient.connect(),
+    redisPublisher.connect(),
+    redisSubscriber.connect(),
+  ]);
+
+  await redisSubscriber.pSubscribe(`${roomChannelPrefix}:*`, handleRoomMessage);
+};
+
+const handleRoomMessage = (rawMessage: string, channelName: string) => {
+  const message: WebSocketResponse = JSON.parse(rawMessage);
+  const roomId = channelName.split(':')[1];
+
+  websocket.sendMessageToRoom(roomId, message);
+};
+
+export const createRoom = async (roomId: string, user: Member) => {
+  console.log(
+    `[redis]: Create room with id=${roomId} for user=${user.username}`
+  );
+  await Promise.all([
+    redisClient.rPush(toRoomKey(roomId), user.username),
+    redisClient.set(toUserKey(user.username), roomId),
```

Comment:
We're using `toRoomKey` and `toUserKey` as a way to separate the types of keys being stored in Redis, since we're using Redis as a simple key-value store